### PR TITLE
Adding sample for OAuth 2.0 DCRM implementation

### PR DESCRIPTION
**Pull request by Abilashini**

### CODE REVIEW

File: modules/samples/oauth2/playground2/src/main/org/wso2/sample/identity/oauth2/OAuth2Constants.java [hunk 1]
```diff
@@ -51,4 +51,26 @@
...
+    public static final String READ = "read";
+    public static final String UPDATE = "update";
+    public static final String DELETE = "delete";
+    public static final String DCRM_METHOD = "dcrmMethod";
+    public static final String CLIENT_ID = "clientId";
+    public static final String CLIENT_SECRET = "clientSecret";
+    public static final String CLIENT_NAME = "clientName";
+    public static final String GRANT_TYPES = "grantTypes";
+    public static final String REDIRECT_URIS = "redirectUris";
+    public static final String CLIENT_CONFIGURATION_ENDPOINT = "https://localhost:9443/identity/register/";
```
Comment (IsuraD) [inline modules/samples/oauth2/playground2/src/main/org/wso2/sample/identity/oauth2/OAuth2Constants.java]: This is the old endpoint. This should be changed to the latest one

File: modules/samples/oauth2/playground2/src/main/org/wso2/sample/identity/oauth2/OAuth2Constants.java [hunk 2]
```diff
@@ -51,4 +51,26 @@
...
+    public static final String AUTHORIZATION = "Basic YWRtaW46YWRtaW4=";
```
Comment (IsuraD) [inline modules/samples/oauth2/playground2/src/main/org/wso2/sample/identity/oauth2/OAuth2Constants.java]: This should be configurable

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    String grantTypes = request.getParameter(OAuth2Constants.GRANT_TYPES);
+    String redirectUris = request.getParameter(OAuth2Constants.REDIRECT_URIS);
+    String deleteConfirmation = null;
+
+    HttpClient client = new DefaultHttpClient();
+    BufferedReader reader;
+    JSONObject jsonObject = null;
+
+    try {
+        if (Objects.equals(method, OAuth2Constants.DELETE)) {
```
Comment (IsuraD) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp]: Why do we use Objects.equals here. 

better to use StringUtils.equals

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+            int statusCode = deleteResponse.getStatusLine().getStatusCode();
+
+            if (statusCode == 204) {
+                deleteConfirmation = "Application has been successfully deleted.";
+            } else {
+                deleteConfirmation = "Application has not been successfully deleted.";
+            }
+        } else {
+            if (Objects.equals(method, OAuth2Constants.CREATE)) {
```
Comment (IsuraD) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp]: Why do we use Objects.equals here.

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+                StringEntity entity = new StringEntity(object.toJSONString());
+                httpPost.setEntity(entity);
+
+                HttpResponse createResponse = client.execute(httpPost);
+                reader = new BufferedReader(new InputStreamReader(createResponse.getEntity().getContent()));
+                Object obj = JSONValue.parse(reader);
+                reader.close();
+                jsonObject = (JSONObject) obj;
+
+            } else if (Objects.equals(method, OAuth2Constants.READ)) {
```
Comment (IsuraD) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp]: Why do we use Objects.equals here.

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+                HttpGet httpGet = new HttpGet(OAuth2Constants.CLIENT_CONFIGURATION_ENDPOINT + clientId);
+                httpGet.addHeader(HttpHeaders.AUTHORIZATION, OAuth2Constants.AUTHORIZATION);
+                HttpResponse readResponse = client.execute(httpGet);
+
+                reader = new BufferedReader(new InputStreamReader(readResponse.getEntity().getContent()));
+                Object obj = JSONValue.parse(reader);
+                reader.close();
+                jsonObject = (JSONObject) obj;
+
+            } else if (Objects.equals(method, OAuth2Constants.UPDATE)) {
```
Comment (IsuraD) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp]: Why do we use Objects.equals here.

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    try {
+        if (Objects.equals(method, OAuth2Constants.DELETE)) {
+            HttpDelete httpDelete = new HttpDelete(OAuth2Constants.CLIENT_CONFIGURATION_ENDPOINT + clientId);
+            httpDelete.addHeader(HttpHeaders.AUTHORIZATION, OAuth2Constants.AUTHORIZATION);
+
+            HttpResponse deleteResponse = client.execute(httpDelete);
+
+            int statusCode = deleteResponse.getStatusLine().getStatusCode();
+
+            if (statusCode == 204) {
```
Comment (IsuraD) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp]: use a constant

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+                    rUriArray.add(str);
+                }
+                object.put(OAuth2Constants.DCRMMetaData.CLIENT_ID, clientId);
+                object.put(OAuth2Constants.DCRMMetaData.CLIENT_SECRET, clientSecret);
+                object.put(OAuth2Constants.DCRMMetaData.CLIENT_NAME, clientName);
+                object.put(OAuth2Constants.DCRMMetaData.GRANT_TYPES, grantTypeArray);
+                object.put(OAuth2Constants.DCRMMetaData.REDIRECT_URIS, rUriArray);
+
+                StringEntity entity = new StringEntity(object.toJSONString());
+
```
Comment (IsuraD) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp]: unnecessary new lines

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+                httpPut.setEntity(entity);
+
```
Comment (IsuraD) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp]: unnecessary new line

File: modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+                jsonObject = (JSONObject) obj;
+
+            } else if (Objects.equals(method, OAuth2Constants.UPDATE)) {
+                HttpPut httpPut = new HttpPut(OAuth2Constants.CLIENT_CONFIGURATION_ENDPOINT + clientId);
+                httpPut.addHeader(HttpHeaders.AUTHORIZATION, OAuth2Constants.AUTHORIZATION);
+                httpPut.addHeader(HttpHeaders.CONTENT_TYPE, OAuth2Constants.CONTENT_TYPE);
+
+                JSONObject object = new JSONObject();
+                JSONArray grantTypeArray = new JSONArray();
+                JSONArray rUriArray = new JSONArray();
```
Comment (IsuraD) [inline modules/samples/oauth2/playground2/src/main/webapp/oauth2-dcrm-request.jsp]: why use  **r**  in the variable? Better to use proper readable variable name

**Comment by JKAUSHALYA** [comment]
Closing until requested changes are done.

**Pull request closed without merging**